Sources/SwiftDocC/Infrastructure/Workspace/DefaultAvailability.swift

```diff
@@ -36,32 +44,82 @@ public struct DefaultAvailability: Codable, Equatable {
         enum CodingKeys: String, CodingKey {
             case platformName = "name"
             case platformVersion = "version"
+            case unavailable = "unavailable"
+        }
+        
+        /// The diferent availability states that can be declared.
+        /// Unavailable or Available with an introduced version.
+        public enum State: Equatable, Hashable, Encodable {
+            case unavailable
+            case available(version: String)
         }
 
         /// The name of the platform, e.g. "macOS".
         public var platformName: PlatformName
+        
+        /// The availability state, e.g unavailable
+        public var state: State
 
         /// A string representation of the version for this platform.
-        public var platformVersion: String
+        @available(*, deprecated, message: "Use `introducedVersion` instead.")
+        public var platformVersion: String {
+            return introducedVersion ?? "0.0"
+        }
+        
+        /// A string representation of the version for this platform
+        /// or nil if it's unavailable.
+        public var introducedVersion: String? {
+            switch state {
+            case .available(let introduced):
+                return introduced.description
+            case .unavailable:
+                return nil
+            }
+        }
 
         /// Creates a new module availability with a given platform name and platform version.
         ///
         /// - Parameters:
         ///   - platformName: A platform name, such as "iOS" or "macOS"; see ``PlatformName``.
-        ///   - platformVersion: A 2- or 3-component version string, such as `"13.0"` or `"13.1.2"`
+        ///   - platformVersion: A 2- or 3-component version string, such as `"13.0"` or `"13.1.2"`.
+        @available(*, deprecated, message: "Use `init(platformName:platformState:) instead.")
```

Comment:
We don't need to deprecate this initializer. It could be used as a way to initialize a value without needing to expose `State` as a public type.

Reply:
Deprecation tag removed 👍